lib/index.js

```diff
@@ -354,11 +354,11 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
     var writeKey = this.options.apiKey;
 
     // Exclude the current domain from the list of servers we're querying
-    var currentTld = getTld(window.location.hostname);
+    var currentTld = topDomain(window.location.hostname);
     var domains = [];
     for (var i=0; i<this.options.crossDomainIdServers.length; i++) {
       var domain = this.options.crossDomainIdServers[i];
-      if (getTld(domain) !== currentTld) {
```

Comment:
I couldn't think of a good way to test this but if you have a suggestion we can.